scripts/rollup/build.js

```diff
@@ -365,139 +332,154 @@ async function createBundle(bundle, bundleType) {
+  }
+  for (let i = 0; i < otherOutputPaths.length; i++) {
+    await asyncCopyTo(mainOutputPath, otherOutputPaths[i]);
+  }
+  console.log(`${chalk.bgGreen.black(' COMPLETE ')} ${logKey}\n`);
+}
+
+function handleRollupWarning(warning) {
+  if (warning.code === 'UNRESOLVED_IMPORT') {
+    console.error(warning.message);
+    process.exit(1);
+  }
+  if (warning.code === 'UNUSED_EXTERNAL_IMPORT') {
+    const match = warning.message.match(/external module '([^']+)'/);
+    if (!match || typeof match[1] !== 'string') {
+      throw new Error(
+        'Could not parse a Rollup warning. ' + 'Fix this method.'
       );
-      console.error(error.message);
-      const {file, line, column} = error.loc;
-      if (file) {
-        // This looks like an error from Rollup, e.g. missing export.
-        // We'll use the accurate line numbers provided by Rollup but
-        // use Babel code frame because it looks nicer.
-        const rawLines = fs.readFileSync(file, 'utf-8');
-        // column + 1 is required due to rollup counting column start position from 0
-        // whereas babel-code-frame counts from 1
-        const frame = codeFrame(rawLines, line, column + 1, {
-          highlightCode: true,
-        });
-        console.error(frame);
-      } else {
-        // This looks like an error from a plugin (e.g. Babel).
-        // In this case we'll resort to displaying the provided code frame
-        // because we can't be sure the reported location is accurate.
-        console.error(error.codeFrame);
-      }
-    } else {
-      console.error(error);
     }
-    process.exit(1);
+    const importSideEffects = Modules.getImportSideEffects();
+    const externalModule = match[1];
+    if (typeof importSideEffects[externalModule] !== 'boolean') {
+      throw new Error(
+        'An external module "' +
+          externalModule +
+          '" is used in a DEV-only code path ' +
+          'but we do not know if it is safe to omit an unused require() to it in production. ' +
+          'Please add it to the `importSideEffects` list in `scripts/rollup/modules.js`.'
+      );
+    }
+    // Don't warn. We will remove side effectless require() in a later pass.
+    return;
   }
+  console.warn(warning.message || warning);
 }
 
-// clear the build directory
-rimraf('build', async () => {
-  try {
-    // create a new build directory
-    fs.mkdirSync('build');
-    // create the temp directory for local npm packing and unpacking
-    // in operating system's default temporary directory
-    fs.mkdirSync(npmPackagesTmpDir);
-    // create the packages folder for NODE+UMD bundles
-    fs.mkdirSync(path.join('build', 'packages'));
-    // create the dist folder for UMD bundles
-    fs.mkdirSync(path.join('build', 'dist'));
+function handleRollupError(error) {
+  loggedErrors.add(error);
+  if (!error.code) {
+    console.error(error);
+    return;
+  }
+  console.error(
+    `\x1b[31m-- ${error.code}${error.plugin ? ` (${error.plugin})` : ''} --`
```

Comment:
Why not use `chalk.red()` ?

Reply:
Copy paste of existing code, didn't look into how it works.

Reply:
Gotcha. Should be identical to `chalk.red('string')`